pkg/operator/storageclasscontroller/vmware.go

```diff
@@ -4,20 +4,56 @@ import (
 	"context"
 	"fmt"
 	"net/url"
+	"regexp"
 	"time"
 
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/klog/v2"
 	"k8s.io/legacy-cloud-providers/vsphere"
 
+	v1 "github.com/openshift/api/config/v1"
 	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/pbm"
+	"github.com/vmware/govmomi/pbm/types"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/mo"
 	"github.com/vmware/govmomi/vim25/soap"
+	vim "github.com/vmware/govmomi/vim25/types"
 )
 
 const (
 	secretName = "vmware-vsphere-cloud-credentials"
+	apiTimeout = 10 * time.Minute
+
+	categoryNameTemplate = "openshift-%s"
+	policyNameTemplate   = "openshift-storage-policy-%s"
+	vim25Prefix          = "urn:vim25:"
 )
 
-func (c *StorageClassController) newClient(ctx context.Context, cfg *vsphere.VSphereConfig, username, password string) (*govmomi.Client, error) {
+var associatedTypesRaw = []string{"StoragePod", "Datastore", "ResourcePool", "VirtualMachine"}
```

Comment:
If I understand it correctly, this is list if types that the *operator* needs to add to a category created by the installer. Should it contain `VirtualMachine` and `ResourcePool`?

If it's the full list of types that OCP may need, then `Folder` is missing.

Reply:
If we do end up creating the category in rare cases - we are going to need full list. So, I am going to copy `folder` over here.